Updating typedef to use different export syntax

smallsaucepan opened the pull request:
HI @rowanwins. Would you mind taking a look at this PR? Trying to update Turf to "esm first" packages, and having problems importing sweepline-intersections now that we've got node16 set as the moduleResolution mode.

It might be related to https://github.com/microsoft/TypeScript/issues/50175#issuecomment-1204986989 in that I'm getting the same initial error message when trying to import sweepline-intersections' default export - ```This expression is not callable ... has no call signatures```

Happy to talk through it including any troubleshooting ideas you might have to try first.

rowanwins commented:
Thanks for flagging @smallsaucepan - don't you love package bundling and importing issues 😂 

A couple of other things come to mind for the `package.json`
- adding a `type` key
- adding an `exports` key

Making those adjustments would bring this package inline with what Turf does for it's sub-packages
https://github.com/Turfjs/turf/blob/master/packages/turf-along/package.json#L26-L42

That said I should possibly review the rollup config, because I probably don't need to include nodeResolve plugin in all the outputs.

I'm certainly no expert on package bundling, a TS inparticular can drive me a bit bonkers, but I'll take a look and see what I can work out. Can you point me at the relevant TurfJS branch where you're getting the issue?

smallsaucepan commented:
Thanks @rowanwins. Packaging and importing is certainly a ... rich vein of potential learnings 🥲

> Can you point me at the relevant TurfJS branch where you're getting the issue?

Sure thing. The branch where this popped up is https://github.com/smallsaucepan/turf/tree/type-module Take a look at turf-kinks. Pretty sure it's because we moved to node16 module resolution.

Have rolled back a couple of changes and changed the tack of the PR too. [arethetypeswrong](https://arethetypeswrong.github.io/?p=sweepline-intersections%401.5.0) produces the below diagnostic and advice:

> The resolved types use export default where the JavaScript file appears to use module.exports =. This will cause TypeScript under the node16 module mode to think an extra .default property access is required, but that will likely fail at runtime. These types should use export = instead of export default.

Would you want to add those other items (type and exports fields) in this PR? Or a separate one?

smallsaucepan commented:
Hmm, don't think that will work either. Can't use ```export = Blah``` if there are other exports. Will move this PR to draft and regroup.

rowanwins commented:
Superceded by https://github.com/rowanwins/sweepline-intersections/pull/19
